internal/formats/common/cyclonedxhelpers/properties.go

```diff
@@ -0,0 +1,64 @@
+package cyclonedxhelpers
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/syft/pkg"
+)
+
+func Properties(p pkg.Package) *[]cyclonedx.Property {
+	props := []cyclonedx.Property{}
+	props = append(props, getCycloneDXProperties(p)...)
+	if len(p.Locations) > 0 {
+		for _, l := range p.Locations {
+			props = append(props, getCycloneDXProperties(l.Coordinates)...)
+		}
+	}
+	if hasMetadata(p) {
+		props = append(props, getCycloneDXProperties(p.Metadata)...)
```

Comment:
We should additionally capture "pkg.MetadataType" as a property to help in future decoding efforts (how to take a bag of fields and generate a syft package when decoding a syft-generated CycloneDX document back into native syft objects)